sunpy/data/data_manager/cache.py

```diff
@@ -130,35 +132,32 @@ def _get_by_url(self, url):
         url : `str`
             URL of the file.
         """
-        details = self._storage.find_by_key('url', url)
-        return details
+        return self._storage.find_by_key('url', url)
 
-    def _download_and_hash(self, urls, namespace=''):
+    def _download_and_hash(self, urls, namespace):
         """
-        Downloads the file and returns the path, hash and url it used to download.
+        Downloads the file and returns the path, hash, and URL it used to download.
 
         Parameters
         ----------
-        urls : `list`
-            List of urls.
+        urls : `list` of `str`
+            List of URLs.
 
         Returns
         -------
-        `str`, `str`, `str`
-            Path, hash and URL of the file.
+        `pathlib.Path`, `str`, `str`
+            Path to the downloaded file, SHA-256 hash, and the URL used.
         """
-        def download(url):
-            path = self._cache_dir / (namespace + get_filename(urlopen(url), url))
-            self._downloader.download(url, path)
-            shahash = hash_file(path)
-            return path, shahash, url
-
         errors = []
         for url in urls:
             try:
-                return download(url)
+                path = self._cache_dir / (namespace + get_filename(urlopen(url), url))
+
+                self._downloader.download(url, path,overwrite = True)
```

Comment:
Why the overwrite?

Reply:
so if any file changes on disk it can be overwritten (since I think the default argument is is false https://github.com/Cadair/parfive/blob/main/parfive/downloader.py) so if a file changes and we have overwrite as False the old file will stay